### PR TITLE
Add CI for docs links

### LINKED ISSUE #1207: ci for checking links
# Description
Thanks for `opsdroid` project made some improvements on my tool, and I keep using your docs live goto docs for tests. One thing I would like to ask to focus on is documentation.  

You already have in place `mkdocs build` instruction, but we have two important conclusions:
1) It's wouldn't error if there something wrong.
2) It shows what problems you already have.

Like at this moment, running `mkdocs build` will show next message 

```
INFO    -  Building documentation...
WARNING -  Config value: 'pages'. Warning: The 'pages' configuration
           option has been deprecated and will be removed in a future release of
           MkDocs. Use 'nav' instead.
INFO    -  Cleaning site directory
INFO    -  The following pages exist in the docs directory, but are not included 
           in the "nav" configuration:
  - ISSUE_TEMPLATE.md
  - PULL_REQUEST_TEMPLATE.md
  - matchers/watson.md
```
well, more or less. 

## Suggestion

Can I add `deadlinks` as checking instruction to travis-ci, so it will error if broken link introduced to docs? I have already tried to add it to tox, but it looks like it requires some extra effort.
I am planning to make 0.0.3 version on the next two weeks (ideally on 15/11), and use it for this proposes. 

Any thoughts?

### PR DESCRIPTION
**Pull request by butuzov**
# Description
Fixes #1207 

Next changes add additional tool and change current processes in order to get better quality for documentation.

1. Running `mkdocs` with `-s`, so it throw `exit 1` code is any case of deprecated features or issues with documentation, or any other warnings.
1. Running `mkdocs` in background in `serve` mode and checking links using https://github.com/butuzov/deadlinks
1. **Important**: checks implemented only to local links (ones that generated by `mkdocs` itself). In order to enable it for external links (not recommended as external links are standalone resources) use it with `-e` option. 

# Examples
This output can be found in Travis build log for `docs` stage.

## Example (Error)
```
=====================================================================
URL=<http://127.0.0.1:3001/>; External Cheks=Off; Threads=10; Retry=3
=====================================================================
Links Total: 231; Found: 44; Not Found: 1; Ignored: 186; Redirects: 0
---------------------------------------------------------------------
[ failed ] http://127.0.0.1:3001/docs/tutorials/introduction-vim.md
ERROR: InvocationError for command /home/travis/build/butuzov/opsdroid/.tox/docs/bin/deadlinks http://127.0.0.1:3001/ -n 10 -r 3 --no-progress --fiff (exited with code 1)
```

## Example (OK)

```bash
=====================================================================
URL=<http://127.0.0.1:3001/>; External Cheks=Off; Threads=10; Retry=3
=====================================================================
Links Total: 230; Found: 44; Not Found: 0; Ignored: 186; Redirects: 0
---------------------------------------------------------------------
``` 

In Additions:
* `pages` renamed to `nav` (in mkdocs.yml) as fix for error found
   while improving ci.

## Status
**READY**  

## Type of change
- Documentation (fix or adds documentation)

# How Has This Been Tested?
34 force pushes to travis from forked repository.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] New and existing unit tests pass locally with my changes

Checks and output of tests available at https://travis-ci.org/butuzov/opsdroid/builds

### CODE REVIEW

**Comment by butuzov** [comment]
Hmm, is there a way to skip a step on appveyor within tox? or I can just switch these tests into Travis

File: requirements_dev.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ alabaster==0.7.12
 commonmark==0.9.1
 recommonmark==0.6.0
 mkdocs==1.0.4
+deadlinks
```
Comment (jacobtomlinson) [inline requirements_dev.txt]: This should be in `requirements_test.txt` as this is being used during the test suite. Also can you please pin to the latest version.

Comment (butuzov) [reply requirements_dev.txt]: Can we have it on hold? I am planning to move it out to "travis ci" only ci tests as a standalone stage. I just have no time atm. as for "pining last version" got it.

**Comment by jacobtomlinson** [comment]
After merging #1265 there were some conflicts here. 

We are now using Sphinx for docs so I've just pulled master into this PR and added the `deadlinks` stuff back in to the docs tests.

**Comment by butuzov** [comment]
Congrats! 

I am going to release 0.2.0 next Monday, which allows running local checks, as far as I see (I am also using `sphinx`) sphinx isn't really nice and not serving files itself.

What we can do now:
1) wait for the release and use pip module 
2) use dev version now (only available as docker image), and replace image `tag` it the near future.

## example of local checks

```
docker run --rm -it --network=host -v "/projects/gobyexample/public:/docs" \
  docker.io/butuzov/deadlinks:dev internal --root /docs
```

it's not as fast as running remote checks  (somehow with multithreading i was expecting better performance), but it works.

**Comment by butuzov** [comment]
Please ignore next force pushes, I am testing "Travis-ci and docker ci check".

**Comment by jacobtomlinson** [comment]
It looks like you've overwritten the changes I pushed here by force pushing 😒.

I'm happy to stick with the current version for now. @pyup-bot will automatically open a PR for us whe you push our a new version.

**Comment by butuzov** [comment]
Yeah. I know. I did it for next reasons:
1) You running your ci via tox at appveyor, honestly... I don't even have where to test on windows and it's failed last time.
2) `0.1.1` Dosn't  have feature I am going to use now (local files checking). 
3) I will add check temporarily as docker image, and change it back to travis ci only or tox only (if you don't mind failing ci on app veyour) one 0.2.0 released.

is it ok for you?

**Comment by butuzov** [comment]
Ok, guess we have some results 
https://travis-ci.org/butuzov/opsdroid/jobs/617718566

**Comment by butuzov** [comment]
I am not going to change source code before release, just to write a bit more docs. Once 0.2.0 I will update the current ci pipeline (just let me know what you think about appveyor and tox integration).

For now - it's ready.

**Comment by butuzov** [comment]
Got it. Sorry for mis understanding and self-management.

**Pull request merged**